packages/expo-splash-screen/android/src/main/java/expo/modules/splashscreen/SplashScreenModule.kt

```diff
@@ -0,0 +1,55 @@
+package expo.modules.splashscreen
+
+import android.content.Context
+
+import org.unimodules.core.ExportedModule
+import org.unimodules.core.ModuleRegistry
+import org.unimodules.core.Promise
+import org.unimodules.core.errors.CurrentActivityNotFoundException
+import org.unimodules.core.interfaces.ActivityProvider
+import org.unimodules.core.interfaces.ExpoMethod
+
+class SplashScreenModule(context: Context) : ExportedModule(context) {
+  companion object {
+    private const val NAME = "ExpoSplashScreen"
+    private const val ERROR_TAG = "ERR_SPLASH"
+  }
+
+  private lateinit var activityProvider: ActivityProvider
+
+  override fun getName(): String {
+    return NAME
+  }
+
+  override fun onCreate(moduleRegistry: ModuleRegistry) {
+    activityProvider = moduleRegistry.getModule(ActivityProvider::class.java)
+  }
+
+  @ExpoMethod
+  fun preventAutoHideAsync(promise: Promise) {
+    val activity = activityProvider.currentActivity
+    if (activity == null) {
+      promise.reject(CurrentActivityNotFoundException())
```

Comment:
please include a code (`ERR_SPLASH_SCREEN_<1-2-word description>`), and a helpful message that explains how to fix the problem, with all of these promise rejections!

Reply:
This one is already done inside `CurrentActivityNotFoundException` 😉

Reply:
In that case, could you verify that `promise.reject` automatically uses the right fields from the exception? I'm especially thinking about the `code` since it doesn't normally exist on exception objects.

Reply:
Yeah, you can see it here:
https://github.com/expo/expo/blob/master/packages/@unimodules/core/android/src/main/java/org/unimodules/core/Promise.java#L5-26
and here:
https://github.com/expo/expo/blob/master/packages/@unimodules/core/android/src/main/java/org/unimodules/core/errors/CurrentActivityNotFoundException.java#L5-14